apps/yggdrasil_web/lib/yggdrasil_web/ensure_permission.ex

```diff
@@ -0,0 +1,77 @@
+defmodule YggdrasilWeb.EnsurePermission do
+  @moduledoc ~S"""
+  ## Overview
+  This plug takes in a connection and set of permission and ensures that all
+  permissions provided are present for the user on the connection.
+
+    import Yggdrasil.EnsurePermission
+
+    plug EnsurePermission, [foo: [:read]] when action in [:index]
+
+  ## Roles and Permissions 
+  The user's permissions are dictated by their role. Each role has a set
+  resources and each resource has a set of permissions. With this scheme
+  the res_perms passed in are in the form of:
+
+  [resource: [perm_set], resource: [perm_set]]
```

Comment:
Did you mean to have this indented so it'll be interpreted as a code block if rendered?